src/map/lua/lua_baseentity.cpp

```diff
@@ -3870,6 +3871,28 @@ inline int32 CLuaBaseEntity::breakLinkshell(lua_State* L)
     return 1;
 }
 
+/************************************************************************
+*  Function: selectDailyItem()
+*  Purpose : Picks a random item for Daily Tally chest
+*  Example : player:selectDailyTally(dial_number)
+*  Notes   : 
+************************************************************************/
+
+inline int32 CLuaBaseEntity::selectDailyItem(lua_State* L)
+{
+    TPZ_DEBUG_BREAK_IF(m_PBaseEntity == nullptr);
+    TPZ_DEBUG_BREAK_IF(m_PBaseEntity->objtype != TYPE_PC);
+    TPZ_DEBUG_BREAK_IF(lua_isnil(L, 1) || !lua_isnumber(L, 1));
+    CCharEntity* PChar = ((CCharEntity*)m_PBaseEntity);
+    int selection = daily::SelectItem((uint8)lua_tointeger(L, 1));
+    while ((itemutils::GetItem(selection)->getFlag() & 0x8000) > 0 && charutils::HasItem(PChar, selection))
+    {
+        selection = daily::SelectItem((uint8)lua_tointeger(L, 1));
+    }
```

Comment:
I'm leaning towards keeping all the logic for selection in a single place, so the binding just passes the Char and dial to `daily::` and returns the final result.

_Ideally_ we'd have the select daily util function be exposed to Lua instead of adding yet another binding to baseentity, but I don't feel strongly enough to _press_ that particular route. 😅

Reply:
I actually started by putting the rerolling directly in daily::SelectItem, but it involved adding a loop for each dial case, which is why I moved it here.